dotnet/src/Connectors/Connectors.Postgres.UnitTests/PostgresVectorStoreRecordMapperTests.cs

```diff
@@ -0,0 +1,202 @@
+    {
+        // Arrange
+        var vector = new ReadOnlyMemory<float>([1.1f, 2.2f, 3.3f, 4.4f]);
+        var storageVector = PostgresVectorStoreRecordPropertyMapping.MapVectorForStorageModel(vector);
+
+        var storageModel = new Dictionary<string, object?>
+        {
+            ["Key"] = "key",
+            ["StringProperty"] = "Value1",
+            ["IntProperty"] = 5,
+            ["FloatVector"] = storageVector
+        };
+
+        var definition = GetRecordDefinition<string>();
+        var propertyReader = GetPropertyReader<TestRecord<string>>(definition);
+
+        var mapper = new PostgresVectorStoreRecordMapper<TestRecord<string>>(propertyReader);
+
+        // Act
+        var result = mapper.MapFromStorageToDataModel(storageModel, new() { IncludeVectors = includeVectors });
+
+        // Assert
+        Assert.Equal("key", result.Key);
+        Assert.Equal("Value1", result.StringProperty);
+        Assert.Equal(5, result.IntProperty);
+
+        if (includeVectors)
+        {
+            Assert.NotNull(result.FloatVector);
+            Assert.Equal(vector.Span.ToArray(), result.FloatVector.Value.Span.ToArray());
+        }
+        else
+        {
+            Assert.Null(result.FloatVector);
+        }
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void MapFromStorageToDataModelWithNumericKeyReturnsValidGenericModel(bool includeVectors)
+    {
+        // Arrange
+        var vector = new ReadOnlyMemory<float>([1.1f, 2.2f, 3.3f, 4.4f]);
+        var storageVector = PostgresVectorStoreRecordPropertyMapping.MapVectorForStorageModel(vector);
+
+        var storageModel = new Dictionary<string, object?>
+        {
+            ["Key"] = (ulong)1,
+            ["StringProperty"] = "Value1",
+            ["IntProperty"] = 5,
+            ["FloatVector"] = storageVector
+        };
+
+        var definition = GetRecordDefinition<ulong>();
+        var propertyReader = GetPropertyReader<TestRecord<ulong>>(definition);
+
+        var mapper = new PostgresVectorStoreRecordMapper<TestRecord<ulong>>(propertyReader);
+
+        // Act
+        var result = mapper.MapFromStorageToDataModel(storageModel, new() { IncludeVectors = includeVectors });
+
+        // Assert
+        Assert.Equal((ulong)1, result.Key);
+        Assert.Equal("Value1", result.StringProperty);
+        Assert.Equal(5, result.IntProperty);
+
+        if (includeVectors)
+        {
+            Assert.NotNull(result.FloatVector);
+            Assert.Equal(vector.Span.ToArray(), result.FloatVector.Value.Span.ToArray());
+        }
+        else
+        {
+            Assert.Null(result.FloatVector);
+        }
+    }
+
+    #region private
+
+    private static VectorStoreRecordDefinition GetRecordDefinition<TKey>()
+    {
+        return new VectorStoreRecordDefinition
+        {
+            Properties = new List<VectorStoreRecordProperty>
```

Comment:
Might be good to add one or two enumerables to the list as well to verify that they word as expected.